modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/dto/v1_0/converter/ObjectEntryDTOConverter.java

```diff
@@ -534,7 +537,11 @@ private ObjectEntry _toDTO(
 					_userLocalService.fetchUser(objectEntry.getUserId()));
 				dateCreated = objectEntry.getCreateDate();
 				dateModified = objectEntry.getModifiedDate();
-				externalReferenceCode = objectEntry.getExternalReferenceCode();
+				externalReferenceCode = SanitizerUtil.sanitize(
+					objectEntry.getCompanyId(), 0, objectEntry.getUserId(),
+					objectDefinition.getClassName(), 0, ContentTypes.TEXT_HTML,
+					Sanitizer.MODE_ALL, objectEntry.getExternalReferenceCode(),
+					null);
```

Comment:
@leilanyaragao I was wondering if we could sanitize the ERC before saving it to the database. This way, the "bad" value is not stored in the database and we perform the sanitize only once. Here are some examples I found in the portal:

- Change in the [portlet-model-hints.xml](https://github.com/liferay/liferay-portal/commit/36b21590d1cac644864b10fbec02c2cac20098b3#diff-636abdb491d9fa97f95784ca5cfedbe8d7700081a95ef30311442b1048c6d13f)
- Change in the [LocalServiceImpl](https://github.com/liferay/liferay-portal/blob/bde9eb760be3e33c3037364b8ecde153eb6678bc/modules/apps/calendar/calendar-service/src/main/java/com/liferay/calendar/service/impl/CalendarBookingLocalServiceImpl.java#L150)

The first option seems more "clean" to me.

/cc @murilostodolni

Reply:
Hey @carolmariaabb, I sanitized it when get because @MarinhoFeliphe  had done something similar to this in [RichText](https://github.com/liferay-objects/liferay-portal/blob/bde9eb760be3e33c3037364b8ecde153eb6678bc/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/RichTextObjectFieldBusinessType.java#L79C22-L79C22).  When I was resolving it, I even spoke to him and we thought it was better to maintain the standard and leave it at the get. 
Should I change it or leave it the way it is to maintain the standard?